### ISSUE TITLE
[TW-1516] Hook input is not ended with <EOF>

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1542]
_Tomas Babej on 2015-01-16T22:43:07Z says:_ 

 When hooks read the input, they do not receive the <EOF>. This will cause halting when reading the stdin from the hooks:

```
for line in sys.stdin:
   <blocks waiting for more>
   ...
```

One could argue that hooks should now how many lines to except, and read only that much (add hooks one line, modify two). However, this is not true for on-exit hooks.

The on-exit hooks will always block, since it can get arbitrary number of lines as input.

Non-blocking reads are much harder, especially when dealing with pipes/portability issues.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-01-16T22:43:07Z
Modified: 2015-01-18T08:37:25Z
```

**Comment by taskwarrior** [comment]
_Wim Schuermann on 2015-01-18T08:37:25Z says:_ 

 Fixed with [d2928dc4e4572cbb351b6df9706fd50a095015a8|https://git.tasktools.org/projects/TM/repos/task/commits/d2928dc4e4572cbb351b6df9706fd50a095015a8]

This was a quick fix that I didn't have time to test properly; since Paul knows this and approved it, lay any blame for issues it creates on him. :)
